sql/src/main/java/io/crate/analyze/expressions/ValueNormalizer.java

```diff
@@ -68,7 +69,7 @@ public static Symbol normalizeInputForReference(Symbol valueSymbol, Reference re
         Literal<?> literal;
         try {
             literal = (Literal<?>) valueSymbol.cast(reference.valueType());
-        } catch (ConversionException e) {
```

Comment:
or shall we go with catching `Exception` as it is done in  [normalizePrimitiveValue]?(https://github.com/crate/crate/pull/9524/files#diff-783af917e859cab7610aaeb51d8258beL169)

Reply:
I think it is better to be more specific when catching the exceptions.

Reply:
@seut because we use `PgArrayParser` in `ArrayType` now, see https://github.com/crate/crate/commit/470eaa5acfa9b29df372d90c4f8d1cc371f512c0#diff-299c839efaea31569151e5ad86ebab25R115

Reply:
ah right, on literal casts, `value()` is called.